README.md

```diff
@@ -1,24 +1,211 @@
-application up and running.
+A basic multi-page rails demo app created to showcase [Shopify's Next-Gen
+Auth][1] using [Turbolinks][2]. This demo shows how to convert multi-page apps
+into Next-Gen Auth enabled apps.
 
-Things you may want to cover:
+If you have a regular multi-page app and you want to use the new Session Token
+Auth but you're unsure or not yet ready to convert your app to an SPA, you can
+still use Session Tokens by converting your app to use Turbolinks. Turbolinks
+is a JavaScript library that makes your application behave as if it were a
+single-paged application.
 
-* Ruby version
+## Getting started
 
-* System dependencies
+> __Important:__ This guide assumes your app has the Next-Gen Auth beta feature
+> enabled. This provides your app early access to use JWT based authorization.
+> Reach out to someone on the App Authorization team if you are unsure whether your
+> app has this enabled.
 
-* Configuration
+If you are looking to install this app on your own testing environment, and
+already have a `Shopify API Key` and `Shopify API Secret` from your partners
+dashboard, you can pull this repository and run the following:
 
-* Database creation
+```console
+$ bundle install
+$ yarn install
+$ rails db:migrate
+```
+See [this section][3] for a guide on setting up your `.env` file.
 
-* Database initialization
+You will also need to use [ngrok][4] to make `localhost:3000` available to the
+internet. You should use the secure HTTPS URL generated by ngrok to create your
+app with on your Partners dashboard.
 
-* How to run the test suite
+To start the app, simply run:
 
-* Services (job queues, cache servers, search engines, etc.)
+```console
+$ rails s
+```
 
-* Deployment instructions
+Your requests to protected resources should be secured with an `Authorization:
+Bearer token` header.
 
-* ...
+![App dashboard][s2]
+
+_**Above:** A sample multi-paged app with an authenticated landing page._
+
+![App requests][s1]
+
+_**Above:** Requests made to protected resources are authenticated using JWTs._
+
+## Enabling Turbolinks on your app
+
+The official [Turbolinks GitHub guide][2] is an excellent resource on getting
+started with Turbolinks. Below are the steps taken to bootstrap Turbolinks on
+this sample app.
+
+1. Add the `turbolinks` gem to your Gemfile:
+
+```rb
+gem 'turbolinks', '~> 5'
+```
+
+2. Run `bundle install`.
+
+3. Add the `turbolinks` package to your application: `yarn add turbolinks`.
+
+4. Add the following line to `app/javascript/packs/application.js` if your app
+uses webpack to manage its manifest files:
+
+```js
+require("turbolinks").start()
+```
+
+## Making authorized requests to protected resources
+
+Before you can navigate to protected resources using Turbolinks, you should
+fetch an authenticated session token. This sample app creates a `/widgets`
+resource that is protected. It requires JWT authorization to read and
+write.
+
```

Comment:
Can you add the first step as updating the `shopify_app.rb` file with `config.allow_jwt_authentication = true` to ensure that jwt authentication is enabled for `shopify_app`?